crates/cheatcodes/spec/src/vm.rs

```diff
@@ -499,43 +497,104 @@ interface Vm {
     #[cheatcode(group = Evm, safety = Unsafe)]
     function readCallers() external returns (CallerMode callerMode, address msgSender, address txOrigin);
 
+    // ----- Arbitrary Snapshots -----
+
+    /// Snapshot capture an arbitrary numerical value by name.
+    /// The group name is derived from the contract name.
+    #[cheatcode(group = Evm, safety = Unsafe)]
+    function snapshotValue(string calldata name, uint256 value) external;
+
+    /// Snapshot capture an arbitrary numerical value by name in a group.
+    #[cheatcode(group = Evm, safety = Unsafe)]
+    function snapshotValue(string calldata group, string calldata name, uint256 value) external;
+
+    // -------- Gas Snapshots --------
+
+
+    /// Snapshot capture the gas usage of the last call by name.
+    #[cheatcode(group = Evm, safety = Unsafe)]
+    function snapshotGas(string calldata name) external;
+
+    /// Snapshot capture the gas usage of the last call by name in a group.
+    #[cheatcode(group = Evm, safety = Unsafe)]
+    function snapshotGas(string calldata group, string calldata name) external;
```

Comment:
`snapshotLastCallGas`?

Reply:
I prefer the shorthand here but open to changing it

Reply:
I think this might be a bit confusing from UX perspective because `startSnapshotGas` records usage _after_ cheatcode invocation, and `snapshotGas` records usage from the past

Reply:
Naming is a bit of a challenge here where we want to prefix the snapshot type e.g. `snapshotGas`

For now added `snapshotGasLastCall` but that is confusing because the cheatcode is `lastCallGas`